snap/snapcraft.yaml

```diff
@@ -231,33 +232,47 @@ parts:
   kafka:
     plugin: nil
     source: https://github.com/canonical/central-uploader/releases/download/kafka_2.13-${SNAPCRAFT_PROJECT_VERSION}-ubuntu0/kafka_2.13-${SNAPCRAFT_PROJECT_VERSION}-ubuntu0-20231121145749.tgz
+    build-packages:
+    - ca-certificates-java
     stage-packages:
     - openjdk-17-jre-headless
     - util-linux
     - libpsl5
     - curl
     override-build: |
       snapcraftctl build
-
       mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka
       mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka
-
       cp -r ./* $SNAPCRAFT_PART_INSTALL/opt/kafka
-    override-prime: |
-      snapcraftctl prime
-      rm -vf usr/lib/jvm/java-17-openjdk-amd64/lib/security/cacerts
-  prometheus-exporter:
+  extra-maven-libs:
     plugin: nil
     after: [kafka]
     override-build: |
-      curl -o jmx_prometheus_javaagent.jar "https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.17.0/jmx_prometheus_javaagent-0.17.0.jar"
-      EXPECTED_SHA="48ab23a7514f9de5d2ca0acbb8ed1573b3c2ecbef7c5dc4d37c4ba125e538199"
-      JAR_SHA=$(sha256sum jmx_prometheus_javaagent.jar | cut -d " " -f1)
-      if [ $JAR_SHA != $EXPECTED_SHA ]; then exit 1; fi
-      cp jmx_prometheus_javaagent.jar $SNAPCRAFT_PART_INSTALL/
-    organize:
-      jmx_prometheus_javaagent.jar: opt/kafka/libs/jmx_prometheus_javaagent.jar
-
+      # path jar version symlink
+      jars=(
+        "io/strimzi kafka-oauth-common 0.14.0 false"
+        "io/strimzi kafka-oauth-client 0.14.0 false"
+        "io/strimzi kafka-oauth-server 0.14.0 false"
+        "io/strimzi kafka-oauth-server-plain 0.14.0 false"
+        "com/nimbusds nimbus-jose-jwt 9.37.1 false"
+        "io/prometheus/jmx jmx_prometheus_javaagent 0.17.0 true"
+      )
+      for entry in "${jars[@]}"; do
+        base=https://repo1.maven.org/maven2
+        path=$(echo "$entry" | awk '{print $1}')
+        jar_name=$(echo "$entry" | awk '{print $2}')
+        version=$(echo "$entry" | awk '{print $3}')
+        symlink=$(echo "$entry" | awk '{print $4}')
+        jar_file="$jar_name-$version.jar"
+        url="$base/$path/$jar_name/$version/$jar_file"
+        curl -O "$url"
+        expected_sha=$(curl "$url.sha1")
```

Comment:
**question** I'm wondering whether we should really download the sha or rather pin it here. The point is that if a malicious agent will manage to patch the JAR, they can also patch the hash, therefore making this check not useful. 

I believe in the initial version we were hardcoding the hash here, which should provide a bit more security

Reply:
updated to use pinned shas